imjoy/runners/subprocess.py

```diff
@@ -175,20 +539,22 @@ def kill_plugin(engine, pid):
         del plugins[pid]
 
 
-async def kill_all_plugins(engine, ssid):
+@sio_on("reset_engine_plugins", namespace=NAME_SPACE)
+async def reset_engine(engine, sid, _):
+    """Handle plugins when reset engine is called."""
+    await kill_all_plugins(engine, sid)
+
+
+async def kill_all_plugins(engine, sid):
     """Kill all plugins."""
     logger = engine.logger
-    on_kill_plugin = engine.conn.sio.handlers[NAME_SPACE]["kill_plugin"]
     plugin_sids = engine.store.plugin_sids
-    tasks = []
-    for sid in list(plugin_sids.keys()):
+    for plugin_info in plugin_sids.values():
         try:
-            await on_kill_plugin(ssid, {"id": plugin_sids[sid]["id"]})
+            await on_kill_plugin(engine, sid, {"id": plugin_info["id"]})
         except Exception as exc:  # pylint: disable=broad-except
             logger.error("%s", exc)
 
-    return asyncio.gather(*tasks)
```

Comment:
This wasn't used, so I removed it. We can enhance this later and use gather again and handle the exception etc.